.circleci/config.yml

```diff
@@ -3,31 +3,54 @@
 
 version: 2.1
 
+orbs:
+  codecov: codecov/codecov@3.2.4
+  macos: circleci/macos@2
+
 # Workflows orchestrate a set of jobs to be run;
 workflows:
   version: 2
-  build-test-deploy:
+  build-test:
     jobs:
-      - build-and-test
-jobs:
+      - validate-code
+      - test-ios:
+          requires:
+            - validate-code
+      - test-tvos:
+          requires:
+            - validate-code
+      - build_xcframework_and_app:
+          requires:
+            - validate-code
+
+commands: 
+  install_dependencies:
+    steps:
+      # restore pods related caches
+      - restore_cache:
+          name: Restoring Gemfile Cache
+          keys:
+            - 1-gems-{{ checksum "Gemfile.lock" }}
 
-  build-and-test:
-    macos:
-      xcode: 13.4.0 # Specify the Xcode version to use
+      # make sure we're on the right version of cocoapods
+      - run:
+          name: Verify Cocoapods Version
+          command: bundle check || bundle install --path vendor/bundle
 
-    steps:
-      - checkout
+      # save cocoapods version gem data
+      - save_cache:
+          name: Saving Gemfile Cache
+          key: 1-gems-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
 
       # restore pods related caches
       - restore_cache:
+          name: Restoring CocoaPods Cache
           keys:
             - cocoapods-cache-v5-{{ arch }}-{{ .Branch }}-{{ checksum "Podfile.lock" }}
```

Comment:
I would suggest rotating the cache version here to `v6` because you are changing what data is cached.